src/daemon.c

```diff
@@ -547,6 +612,46 @@ int
 hsk_daemon_open(hsk_daemon_t *daemon, hsk_options_t *opt) {
   int rc = HSK_SUCCESS;
 
+  if (opt->checkpoint && HSK_CHECKPOINT != NULL) {
+    // Read the hard-coded checkpoint
+    uint8_t *data = (uint8_t *)HSK_CHECKPOINT;
+    size_t data_len = HSK_STORE_CHECKPOINT_SIZE;
+    if (!hsk_store_inject_checkpoint(&data, &data_len, &daemon->pool->chain)) {
+      fprintf(stderr, "unable to inject hard-coded checkpoint\n");
+      return HSK_EBADARGS;
+    }
+  }
+
+  if (opt->prefix) {
+    if (!hsk_store_exists(opt->prefix)) {
+      fprintf(stderr, "prefix path does not exist\n");
+      return HSK_EBADARGS;
+    }
+
+    // Prefix must have enough room for filename
+    if (strlen(opt->prefix) + 32 >= HSK_STORE_PATH_MAX) {
```

Comment:
Might as well use const for 32 as well or  `hsk_store_filename_len(char *prefix, uint32_t height)`. Define is fine.

Reply:
added in d42ff693566c8d7a56dc7ca44a261199390c2fa3